internal/presenters/presenter_sarif_results_pretty.go

```diff
@@ -0,0 +1,208 @@
+package presenters
+
+import (
+	"bytes"
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/snyk/code-client-go/sarif"
+
+	"github.com/snyk/go-application-framework/pkg/local_workflows/code_workflow"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/json_schemas"
+)
+
+type Finding struct {
+	ID            string
+	Severity      string
+	SeverityLevel int
+	Title         string
+	Message       string
+	Path          string
+	Line          int
+}
+
+type FindingsSummary struct {
+	High   int
+	Medium int
+	Low    int
+}
+
+func convertSarifToFindingsList(input sarif.SarifDocument) []Finding {
+	var findings []Finding
+	for _, run := range input.Runs {
+		for _, result := range run.Results {
+			severity := "n/a"
+			severityLevel := 0
+			var rule sarif.Rule
+			for _, ruleItem := range run.Tool.Driver.Rules {
+				if ruleItem.ID == result.RuleID {
+					rule = ruleItem
+					break
+				}
+			}
+
+			if result.Level == "note" {
+				severity = "LOW"
+			} else if result.Level == "warning" {
+				severity = "MEDIUM"
+				severityLevel = 1
+			} else if result.Level == "error" {
+				severity = "HIGH"
+				severityLevel = 2
+			}
+
+			var title string
+			if rule.ShortDescription.Text != "" {
+				title = rule.ShortDescription.Text
+			} else {
+				title = rule.Name
+			}
+
+			location := sarif.Location{}
+			if len(result.Locations) > 0 {
+				location = result.Locations[0]
+			}
+
+			findings = append(findings, Finding{
+				ID:            result.RuleID,
+				SeverityLevel: severityLevel,
+				Severity:      severity,
+				Title:         title,
+				Path:          location.PhysicalLocation.ArtifactLocation.URI,
+				Line:          location.PhysicalLocation.Region.StartLine,
+				Message:       result.Message.Text,
+			})
+		}
+	}
+	return findings
+}
+
+type TestMeta struct {
+	OrgName  string
+	TestPath string
+}
+
+func PresenterSarifResultsPretty(input sarif.SarifDocument, meta TestMeta) (string, error) {
+	findings := convertSarifToFindingsList(input)
+	summary := code_workflow.CreateCodeSummary(&input)
+
+	str := fmt.Sprintf(`
+Testing %s ...
+%s
+%s
+%s
+`,
+		meta.TestPath,
+		renderFindings(SortFindings(findings)),
+		presenterSummary(summary, meta),
+		getTip(),
+	)
+
+	return str, nil
+}
+
+func renderFindings(findings []Finding) string {
+	if len(findings) == 0 {
+		return ""
+	}
+
+	response := "\nOpen Issues\n\n"
+
+	for _, finding := range findings {
+		response += fmt.Sprintf(` %s %s
+   Path: %s, line %d
+   Info: %s
+
+`, renderInSeverityColor(finding.Severity, fmt.Sprintf("✗ [%s]", finding.Severity)), renderBold(finding.Title), finding.Path, finding.Line, finding.Message)
+	}
+
+	return response
+}
+
+func getTip() string {
+	return `💡 Tip
+
+To view ignored issues, use the --include-ignores option. To view ignored issues only, use the --only-ignores option.`
+}
+
+func presenterSummary(summary *json_schemas.TestSummary, meta TestMeta) string {
```

Comment:
nitpick: for consistency, should it be `renderSummary()`?

Reply:
Fix: b5f851690684c708eb69a42d84d6d1e323a4e6d4